awssql/plugins/developer_connection_plugin.go

```diff
@@ -0,0 +1,155 @@
+/*
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+package plugins
+
+import (
+	"database/sql/driver"
+	"log/slog"
+
+	"github.com/aws/aws-advanced-go-wrapper/awssql/driver_infrastructure"
+	"github.com/aws/aws-advanced-go-wrapper/awssql/error_util"
+	"github.com/aws/aws-advanced-go-wrapper/awssql/host_info_util"
+	"github.com/aws/aws-advanced-go-wrapper/awssql/plugin_helpers"
+	"github.com/aws/aws-advanced-go-wrapper/awssql/plugins/error_simulator"
+	"github.com/aws/aws-advanced-go-wrapper/awssql/utils"
+)
+
+type DeveloperConnectionPluginFactory struct {
+}
+
+func NewDeveloperConnectionPluginFactory() driver_infrastructure.ConnectionPluginFactory {
+	return &DeveloperConnectionPluginFactory{}
+}
+
+func (d *DeveloperConnectionPluginFactory) GetInstance(
+	pluginService driver_infrastructure.PluginService, props *utils.RWMap[string, string]) (driver_infrastructure.ConnectionPlugin, error) {
+	return NewDeveloperConnectionPlugin(pluginService, props), nil
+}
+
+func (d *DeveloperConnectionPluginFactory) ClearCaches() {
+}
+
+type DeveloperConnectionPlugin struct {
+	BaseConnectionPlugin
+	errorSimulatorMethodCallback error_simulator.ErrorSimulatorMethodCallback
+	props                        *utils.RWMap[string, string]
+	nextMethodName               string
+	nextError                    error
+	pluginService                driver_infrastructure.PluginService
+}
+
+func NewDeveloperConnectionPlugin(pluginService driver_infrastructure.PluginService, props *utils.RWMap[string, string]) driver_infrastructure.ConnectionPlugin {
+	return &DeveloperConnectionPlugin{
+		pluginService: pluginService,
+		props:         props,
+	}
+}
+
+func (d DeveloperConnectionPlugin) GetPluginCode() string {
```

Comment:
Some of the methods signatures for the DeveloperConnectionPlugin are pointer receivers and some are value receivers. 
IIRC (I need to double check this), the convention is that we dont have a mix of them. A type should only either have pointer receivers or value receivers. Not both.